### ISSUE TITLE
Crontab : [XCUITest] Error: Command 'idevice_id -l' errored out: Error: spawn idevice_id ENOENT

### ISSUE THREAD

**Issue by SatishKumarKV** [appium/appium#7220]
## The problem

Using Crontab to schedule my TestNG XML containing  suite of class files.
Getting spawn idevice_id ENOENT error...

## Environment

* Appium version (or git revision) that exhibits the issue: 1.6
* Last Appium version that did not exhibit the issue (if applicable): 1.6
* Desktop OS/version used to run Appium: Sierra
* Node.js version (unless using Appium.app|exe): 6.9.1
* Mobile platform/version under test: iOS 10
* Real device or emulator/simulator: Real device
* Appium CLI or Appium.app|exe: CLI

## Details
Using Crontab to schedule my TestNG XML containing  suite of class files.
Getting spawn idevice_id ENOENT error...

## Link to Appium logs
```
[debug] [XCUITest] XCUITestDriver version: 2.0.30
[BaseDriver] Session created with session id: 6f07bbc8-0e20-4267-87c3-1b1b03cf8cb4
[debug] [XCUITest] Xcode version set to '8.0'
[debug] [XCUITest] iOS SDK Version set to '10.0'
[XCUITest] Error: Command 'idevice_id -l' errored out: Error: spawn idevice_id ENOENT
    at exports._errnoException (util.js:1026:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at ChildProcess.<anonymous> (lib/teen_process.js:32:14)
    at emitOne (events.js:96:13)
    at ChildProcess.emit (events.js:188:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:213:12)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
 Error: Command 'idevice_id -l' errored out: Error: spawn idevice_id ENOENT
    at exports._errnoException (util.js:1026:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at ChildProcess.<anonymous> (lib/teen_process.js:32:14)
    at emitOne (events.js:96:13)
    at ChildProcess.emit (events.js:188:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:213:12)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
[debug] [XCUITest] Running iOS simulator reset flow
[MJSONWP] Encountered internal error running command: Error: Command 'idevice_id -l' errored out: Error: spawn idevice_id ENOENT
    at exports._errnoException (util.js:1026:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
    at ChildProcess.<anonymous> (lib/teen_process.js:32:14)
    at emitOne (events.js:96:13)
    at ChildProcess.emit (events.js:188:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:213:12)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
[HTTP] <-- POST /wd/hub/session 500 181 ms - 521 
org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. Original error: Command 'idevice_id -l' errored out: Error: spawn idevice_id ENOENT
    at exports._errnoException (util.js:1026:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:359:16)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 2.49 seconds
Build info: version: '2.48.2', revision: '41bccdd', time: '2015-10-09 19:59:12'
System info: host: 'Satishs-MAC.local', ip: '10.104.147.84', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.12', java.version: '1.8.0_112'
Driver info: io.appium.java_client.ios.IOSDriver
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:206)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:647)
	at io.appium.java_client.DefaultGenericMobileDriver.execute(DefaultGenericMobileDriver.java:42)
	at io.appium.java_client.AppiumDriver.execute(AppiumDriver.java:1)
	at io.appium.java_client.ios.IOSDriver.execute(IOSDriver.java:1)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:247)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:129)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:142)
	at io.appium.java_client.DefaultGenericMobileDriver.<init>(DefaultGenericMobileDriver.java:37)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:161)
	at io.appium.java_client.AppiumDriver.<init>(AppiumDriver.java:180)
	at io.appium.java_client.ios.IOSDriver.<init>(IOSDriver.java:69)
	at com.InitializeEnvironment.IOSAppiumConnection.getDriver(IOSAppiumConnection.java:49)
	at com.ios.testController.iOSUpgradeAppiumTestController.beforeSuite(iOSUpgradeAppiumTestController.java:40)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:510)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:211)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:138)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:296)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
	at org.testng.TestNG.run(TestNG.java:1018)
	at org.testng.TestNG.privateMain(TestNG.java:1325`
```

**Comment by imurchie** [comment]
Does your cron user have `idevice_id` in its PATH?

**Comment by SatishKumarKV** [comment]
Yes @imurchie , Below is the command..
18 18 * * * . ~/.bashrc;PATH=$PATH:/usr/local/bin cd "AutomationFolder" && sh Script.sh

**Comment by imurchie** [comment]
And with the user that cron runs as you can run `idevice_id -l`?

**Comment by SatishKumarKV** [comment]
Placed CD and TestNG inside a perl file .. Not seeing idevice_id anymore. 
But seeing some other errors around Xcode shell script while trying to build 

[debug] [XCUITest] Beginning test with command '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/bin/run-xcodebuild.sh --project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj --scheme WebDriverAgentRunner --destination id=d8814950652da486ef218e96adcf098b881e0be6' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'
[XCUITest] Waiting for WebDriverAgent to start on device
[debug] [XCUITest] Log file for xcodebuild test: /Users/Satish/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/6A1658BF-5C52-4BD5-BBD7-EB6A064781AD/Session-WebDriverAgentRunner-2016-11-11_231011-dEOS5R.log
[Xcode] 2016-11-11 23:10:11.511 xcodebuild[93888:4913268] Error Domain=IDETestOperationsObserverErrorDomain Code=3 "Test operation was canceled. If you believe this error represents a bug, please attach the log file at /Users/Satish/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/6A1658BF-5C52-4BD5-BBD7-EB6A064781AD/Session-WebDriverAgentRunner-2016-11-11_231011-dEOS5R.log" UserInfo={NSLocalizedDescription=Test operation was canceled. If you believe this error represents a bug, please attach the log file at /Users/Satish/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/6A1658BF-5C52-4BD5-BBD7-EB6A064781AD/Session-WebDriverAgentRunner-2016-11-11_231011-dEOS5R.log}

[Xcode] No signing certificate "iOS Development" found:  No "iOS Development" signing certificate matching team ID "KNT489GQP6" with a private key was found.

[Xcode] 

[Xcode] 
Testing failed:
	No signing certificate "iOS Development" found:  No "iOS Development" signing certificate matching team ID "KNT489GQP6" with a private key was found.
** TEST FAILED **


The following build commands failed:
	Check dependencies
(1 failure)

[XCUITest] xcodebuild exited with code '65' and signal 'null'

**Comment by rkavalap** [comment]
Errors not related to appium itself so closing it.

**Comment by ChiWhiteSox56** [comment]
@SatishKumarKV - I'm having an issue that's very similar to the one you're having: https://discuss.appium.io/t/xcodebuild-exited-with-code-65-no-signing-certificate-ios-development-found/14095/6

The only difference that I see is I'm not using Crontab.

Did you find a solution to this problem? If so, will you kindly share it with me? I haven't been able to run tests on iOS real devices since the upgrade to XCUITest and it's critically blocking my work. The link that I shared above lists everything that I've tried to fix this issue.
